miden-lib/src/faucets/mod.rs

```diff
@@ -0,0 +1,73 @@
+use crate::assembler::assembler;
+use miden_objects::{
+    accounts::{Account, AccountCode, AccountId, AccountStorage, AccountType, AccountVault},
+    assembly::ModuleAst,
+    crypto::{dsa::rpo_falcon512, merkle::MerkleStore},
+    utils::vec,
+    AccountError, Word, ZERO,
+};
+
+pub enum AuthScheme {
+    RpoFalcon512 { pub_key: rpo_falcon512::PublicKey },
+}
+
+/// Creates a new faucet account with basic faucet interface, specified authentication scheme,
+/// and provided meta data (token symbol, decimals, max supply).
+///
+/// The basic faucet interface exposes two procedures:
+/// - `distribute`, which mints an assets and create a note for the provided recipient.
+/// - `burn`, which burns the provided asset.
+///
+/// `distribute` requires authentication. The authentication procedure is defined by the specified
+/// authentication scheme.
+pub fn create_basic_faucet(
+    init_seed: [u8; 32],
+    meta_data: Word,
+    auth_scheme: AuthScheme,
+) -> Result<(Account, Word), AccountError> {
+    let (_auth_scheme_prefix, _auth_scheme_procedure, storage_slot_0): (&str, &str, Word) =
```

Comment:
nit: remove the variables that are current not in use.
```suggestion
    let storage_slot_0: (&str, &str, Word) =
```